src/server.rs

```diff
@@ -703,7 +703,6 @@ async fn sync_and_watch_config_dir() {
                                 log::error!("sync config to root failed: {}", e);
                                 match crate::ipc::connect(1000, "_service").await {
                                     Ok(mut _conn) => {
-                                        conn = _conn;
```

Comment:
Really? This creates a bug. That's why I hate refactory.

crate::ipc::connect creates a new connection.

Reply:
You should turn on `[deny(warnings)]` so that the code can be correct without warnings in the first place. This is a very common practice.

Reply:
Here's why I removed that line:

<img width="662" height="172" alt="image" src="https://github.com/user-attachments/assets/3af05828-63da-41fa-926e-e53af856e624" />

Is this just a bug in the analyzer, then?

Reply:
You are right. Here is a bug, the `break` should be  removed.